docs/datamodel/understanding-datamodels.rst

```diff
@@ -67,8 +67,21 @@ Building a data model involves the following tasks:
 - Choose a persistent identifier scheme.
 - Configure the REST API and UI.
 
+Internal vs External representation:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The above mentioned tasks can be divided into two: Internal and external representation. 
+
+The two first ones, JSONSchemas and Elasticsearch mappings, correspond to the internal representation of the data. JSONSchemas are used to validate the data, verifying that it matches the Elasticsearch mapping. These last ones are used to specify the format in which the data is stored in Elasticsearch.
+
+Loaders and serializers do their part on the external representation. The firts ones transform the data received into the internal representation, whereas the serializers do the opposite, they transform the data from its internal representation into the external one. It is recommended to use ``Marshamallow`` to implement the loaders and serializers.
```

Comment:
We could include a short example of the internal representation and external representation (side to side for example) as they are the subject of this part, and it might make it easier for readers to understand how they look like, and the transformations that are happening to move from one to the other.

Reply:
I added a note mentioning that the examples are shown afterwards in each section. Minimum examples are shown in each of them, and it is a bit difficult to put them again here in the introduction without duplicating them and making it repetitive for the user.